modules/tensor_mechanics/include/materials/DiscreteRRIsotropicPlasticity.h

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef DISCRETERRISOTROPICPLASTICITY_H
+#define DISCRETERRISOTROPICPLASTICITY_H
+
+#include "DiscreteRadialReturnStressIncrement.h"
+#include "PiecewiseLinear.h"
+#include "MooseMesh.h"
+
+// forward declarations
+class PiecewiseLinear;
+
+/**
+ */
+
+class DiscreteRRIsotropicPlasticity : public DiscreteRadialReturnStressIncrement
+{
+public:
+  DiscreteRRIsotropicPlasticity( const InputParameters & parameters);
+
+protected:
+  void resetQpProperties(); // Values here need to be set to a constant, ideally zero
+
+  virtual void computeStressInitialize(Real effectiveTrialStress);
+  virtual Real computeResidual(Real effectiveTrialStress, Real scalar);
+  virtual Real computeDerivative(Real effectiveTrialStress, Real scalar);
+  virtual void iterationFinalize(Real scalar);
+  virtual void computeStressFinalize(const RankTwoTensor & plasticStrainIncrement);
```

Comment:
Andrew,

  Is there a reason you’re sending these to me?

  Thanks,

  John

/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\x/x\
John E. Dolbow, Professor
Bass Fellow
Editor, Finite Elements in Analysis
   and Design
Director of Graduate Studies
Civil and Environmental Engineering,
   Mechanical Engineering and Materials
   Science, and Mathematics
Duke University
121 Hudson Hall (mailing address)
172 Hudson Hall (office)
Durham, NC 27708-0287
office: 919-660-5202
email: jdolbowATdukeDOTedu
web: http://www.duke.edu/~jdolbow
lab: http://dcml.pratt.duke.edu/
/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\/x\x/x\

On Mar 17, 2016, at 9:33 AM, Andrew E Slaughter <notifications@github.com<mailto:notifications@github.com>> wrote:


In modules/tensor_mechanics/include/materials/DiscreteRRIsotropicPlasticity.h<https://github.com/idaholab/moose/pull/6602#discussion_r56503370>:

> +/**
> + */
> +
> +class DiscreteRRIsotropicPlasticity : public DiscreteRadialReturnStressIncrement
> +{
> +public:
> +  DiscreteRRIsotropicPlasticity( const InputParameters & parameters);
> +
> +protected:
> +  void resetQpProperties(); // Values here need to be set to a constant, ideally zero
> +
> +  virtual void computeStressInitialize(Real effectiveTrialStress);
> +  virtual Real computeResidual(Real effectiveTrialStress, Real scalar);
> +  virtual Real computeDerivative(Real effectiveTrialStress, Real scalar);
> +  virtual void iterationFinalize(Real scalar);
> +  virtual void computeStressFinalize(const RankTwoTensor & plasticStrainIncrement);


@idaholab/moose-users<https://github.com/orgs/idaholab/teams/moose-users> How strict should we be on doxygen comments? Generally, I personally like to see everything in the header have a doxygen comment, even if it is obvious. However, I am not sure that this is something that should be enforced universally, what are peoples feelings?

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly or view it on GitHub<https://github.com/idaholab/moose/pull/6602/files/5e2f395c45979d61f75e72845fa277986429ab2c#r56503370>